### PR TITLE
WS-12 Implementing basic security

### PR DESCRIPTION
**Pull request by klezner**
Access to /location endpoint only with ADMIN role works.

### CODE REVIEW

**Review by MichalKulygin — APPROVED** [review]

File: src/main/java/pl/kl/weatherservice/WeatherServiceApplication.java [hunk 1]
```diff
@@ -1,13 +1,30 @@
...
+
+    private final UserRepository userRepository;
+    private final PasswordEncoder passwordEncoder;
 
     public static void main(String[] args) {
         SpringApplication.run(WeatherServiceApplication.class, args);
     }
 
+    @Override
+    public void run(String... args) throws Exception {
```
Comment (MisQl) [inline src/main/java/pl/kl/weatherservice/WeatherServiceApplication.java]: let's try to move it to a new class and create it as spring component with specific profile like 'develop'

**Comment by klezner** [comment]
Done.

File: src/main/java/pl/kl/weatherservice/location/LocationController.java [hunk 2]
```diff
@@ -19,6 +20,7 @@
     private final LocationService locationService;
 
     @PostMapping()
+    @PreAuthorize("hasRole('ADMIN')")
```
Comment (MisQl) [inline src/main/java/pl/kl/weatherservice/location/LocationController.java]: extract ADMIN string to constant value or enum

**Review by MisQl — APPROVED** [review]

**Comment by klezner** [comment]
value extracted to app properties

File: src/main/resources/application.properties [hunk 3]
```diff
@@ -3,3 +3,4 @@ spring.datasource.username=
 spring.datasource.password=
 spring.jpa.hibernate.ddl-auto=update
 spring.profiles.active=DEV
+weatherservice.addlocation.permissionrole=ADMIN
```
Comment (MisQl) [inline src/main/resources/application.properties]: do you want to change a name value of this role? if not, then just create a class with a static constant value

File: src/main/java/pl/kl/weatherservice/location/LocationController.java [hunk 4]
```diff
@@ -20,7 +20,7 @@
     private final LocationService locationService;
 
     @PostMapping()
-    @PreAuthorize("hasRole('${weatherservice.addlocation.permissionrole}')")
+    @PreAuthorize("hasRole('Roles.ADD_LOCATION_PERMISSION_ROLE')")
```
Comment (MisQl) [inline src/main/java/pl/kl/weatherservice/location/LocationController.java]: just ADMIN

**Review by MisQl — APPROVED** [review]

**Comment by klezner** [comment]
Fixed

**Pull request merged**